tests/integration/models/reddit/test_rules.py

```diff
@@ -0,0 +1,139 @@
+from unittest import mock
+import pytest
+
+from praw.models import Rule
+
+from ... import IntegrationTest
+
+
+class TestRule(IntegrationTest):
+    @property
+    def subreddit(self):
+        return self.reddit.subreddit(pytest.placeholders.test_subreddit)
+
+    def test_add_rule(self):
+        self.reddit.read_only = False
+        with self.recorder.use_cassette("TestRule.test_add_rule"):
+            rule = self.subreddit.rules.add(
+                "PRAW Test",
+                "all",
+                description="Test by PRAW",
+                violation_reason="PTest",
+            )
+            assert rule.short_name == "PRAW Test"
+            assert rule.kind == "all"
+            assert rule.description == "Test by PRAW"
+            assert rule.violation_reason == "PTest"
+
+    def test_add_rule_without_violation_reason(self):
+        self.reddit.read_only = False
+        with self.recorder.use_cassette(
+            "TestRule.test_add_rule_without_violation_reason"
+        ):
+            rule = self.subreddit.rules.add("PRAW Test 2", "comment")
+            assert rule.short_name == "PRAW Test 2"
+            assert rule.kind == "comment"
+            assert rule.description == ""
+            assert rule.violation_reason == "PRAW Test 2"
+
+    @mock.patch("time.sleep", return_value=None)
+    def test_delete_rule(self, _):
+        self.reddit.read_only = False
+        with self.recorder.use_cassette("TestRule.test_delete_rule"):
+            rule = self.subreddit.rules[-1]
+            rule.delete()
+
+    def test_get_rules(self):
```

Comment:
perhaps better named `test_iter_rules`? definitely not a big deal.

Reply:
That does make more sense, going along with test naming schemes. This does require all of the cassettes to be renamed, so I think it will be better off to use an IDE.